plugins/generator-1.16.4/forge-1.16.4/templates/plant.java.ftl

```diff
@@ -421,6 +421,62 @@ import net.minecraft.block.material.Material;
             </#if>
         </#if>
 
+		<#if (data.canBePlacedOn?size > 0) || hasCondition(data.placingCondition)>
+			<#if (data.canBePlacedOn?size > 0) && data.plantType != "growapable">
+			@Override public boolean isValidGround(BlockState state, IBlockReader worldIn, BlockPos pos) {

```

Comment:
Shouldn't valid ground check for additional condition too?

Reply:
well it already does. The check `hasCondition(data.placingCondition)` is above, on line 424.

Reply:
It does, but then condition return value is not used inside isValidGround

Reply:
Because there is no need to. The `isValidGround` is "working" with the `isValidPosition` so if `isValidPosition` returns true, the blocks used in `isValidGround` will be used. If it returns false, the `isValidGround` will be ignored.

Reply:
isValidGround is also used by Block class by canBeFeritilized which is outside isValidPosition

Reply:
@AlsoSomeoneElse what do you think about that?

Reply:
You could move the additional condition to `isValidGround` for static/double plants, and pass y + 1 instead of y as the dependency (so the check is done on the plant position rather than the ground position). Leave the condition inside `isValidPosition` for growable plants since they don't have a `isValidGround` method

Reply:
I don't get what you mean @AlsoSomeoneElse could you do it plz?

Reply:
I don't think this was resolved, so why was this convo closed?

Reply:
Well you said the PR was ready to merge? 
> Well I did not think other pr would cause them, but that pr was ready for merging

Reply:
I said the other PR that caused conflicts here. This PR is not ready as so far I don't have an impression the valid ground issue/problem was resolved.

Reply:
@Max094Reikeb I opened a PR for your branch that should fix this issue